Libraries/LibCrypto/Hash/OpenSSLHashFunction.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024, Altomani Gianluca <altomanigianluca@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteString.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/Types.h>
+#include <LibCrypto/Hash/HashFunction.h>
+
+#include <openssl/evp.h>
+
+namespace Crypto::Hash {
+
+template<typename T, size_t BlockS, size_t DigestS, typename DigestT = Digest<DigestS>>
+class OpenSSLHashFunction : public HashFunction<BlockS, DigestS, DigestT> {
+    AK_MAKE_NONCOPYABLE(OpenSSLHashFunction);
+
+public:
+    using HashFunction<BlockS, DigestS, DigestT>::update;
+
+    static NonnullOwnPtr<T> create()
+    {
+        auto* context = EVP_MD_CTX_new();
+        return make<T>(context);
+    }
+
+    explicit OpenSSLHashFunction(EVP_MD const* type, EVP_MD_CTX* context)
+        : m_type(type)
+        , m_context(context)
+    {
+        OpenSSLHashFunction::reset();
+    }
+
+    virtual ~OpenSSLHashFunction() override
+    {
+        EVP_MD_CTX_free(m_context);
+    }
+
+    virtual ByteString class_name() const override = 0;
+
+    void update(u8 const* input, size_t length) override
+    {
+        VERIFY(EVP_DigestUpdate(m_context, input, length));
```

Comment:
Please don't put things with side effects inside VERIFY.
It won't be disabled in any case, but it trips some mental alarms related to assert()

Reply:
Done